api/util/mappers/engine.py

```diff
@@ -419,14 +425,29 @@ def _get_prioritised_feature_states(
 def _get_segment_feature_states(
     segments: Iterable["Segment"],
     environment_id: int,
+    latest_environment_feature_versions: (
+        Iterable["EnvironmentFeatureVersion"] | None
+    ) = None,
 ) -> Dict[int, List["FeatureState"]]:
     feature_states_by_segment_id = {}
+    latest_environment_feature_versions = latest_environment_feature_versions or []
```

Comment:
remove along with https://github.com/Flagsmith/flagsmith/pull/3974/files#r1604807951

Reply:
Convert to set for faster membership checks